### PR TITLE
T1 1 create register form

### PR DESCRIPTION
**Pull request by diegoplatap**
# Create Register Form (Users of Foundations)

Users and foundation are able to register in the app when they navigate to the register page **(RegisterPages.js)**

### Key points: 

- The form  was made with React hook forms,  in which is possible to handle  submit , validation (errors) and compare values like the password confimation field. 
-  New button component was created to use it in other sections of the app. **(Primary.Button.js)**
- The form was tested using axios with a  201 status.
- The validations happen before the form is submitted

### Form preview: 
image

### Validation errors:
image

### CODE REVIEW

**Comment by juanan150** [comment]
Lo veo muy bien!
Lo único que le haría sería bajar los mensajes de error un poco, o inclusive ponerlo entre el label y el input, porque en este momento están más cercanos al input superior que al que hacen referencia, entonces tal vez pueda generar confusión, pero de resto muy bacano!

File: src/pages/RegisterPage.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            placeholder="Email"
+            name="email"
+            className="form__field"
+            {...register("email", {
+              required: true,
+              pattern: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/,
+            })}
+          />
+
+          {/* { {errors?.address?.type === 'required' && <p>*Address is required</p>}
```
Comment (leoneloliveros) [inline src/pages/RegisterPage.js:92]: Please remove this commented code

File: src/pages/RegisterPage.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    handleSubmit,
+    watch,
+  } = useForm({ mode: "onBlur" });
+
+  const password = useRef({});
+  password.current = watch("password", "");
+
+  const onSubmit = (data, e) => {
+    e.target.reset();
+    console.log(data);
```
Comment (leoneloliveros) [inline src/pages/RegisterPage.js:20]: Please remove this console.log

File: src/pages/RegisterPage.js (hunk 1 above)
Comment (diegoplatap) [reply src/pages/RegisterPage.js]: Thank you Leonel! I removed unnecessary commented code.

File: src/pages/RegisterPage.js (hunk 2 above)
Comment (diegoplatap) [reply src/pages/RegisterPage.js:21]: Thank you Leonel! I removed unnecessary console log.

**Review by leoneloliveros — APPROVED** [review]
approved!

**Pull request merged**